ADR/0008-integration-service-api-contracts.md

```diff
@@ -0,0 +1,256 @@
+
+## Decision
+
+### Results and Records
+
+The output of each Tekton task will be provided in two forms:
+- As a minimized [Tekton result](https://tekton.dev/docs/pipelines/tasks/#emitting-results) in JSON format listing all test failures
+  - The name of the result will be **HACBS_TEST_OUTPUT**
+- As a Test output JSON file with full test information that’s saved in the Tekton Pipeline Workspace
+  - The name of the file will be in snake case and will be in the form of **test_name_output.json**
+
+The maximum size of a [Task's Results](https://tekton.dev/vault/pipelines-v0.17.3/tasks/#emitting-results) is limited by the container [termination message](https://kubernetes.io/docs/tasks/debug/debug-application/determine-reason-pod-failure/#customizing-the-termination-message) feature of Kubernetes.
+
+App Studio builds are structured as [a shared Persistent Volume per App Studio Workspace](https://docs.google.com/document/d/1IPlihVjkJ4Kb9tdhsk7iz3bn5rkT_SvJCNQhyXzK3aI/edit#bookmark=id.gefgys3vno2). This allows teams to share builds, implement caching and other shared volumes. A single persistent volume is mapped to each default build pipeline. Builds are passed a directory specific to their builds.
+
+#### Tekton Result Format
+
+The Test output of the Tekton result will be a JSON object that includes the context about the test along with the list of check names for all failed checks
+
+The output will provide the following information about the overall test result:
+- **result** - The outcome of the testing task, can be `SUCCESS`, `FAILURE`, `WARNING`, `SKIPPED` or `ERROR`
+- **namespace** - Optional, the rego namespace of the test policy
+- **timestamp** - An UNIX epoch timestamp of the test completion time
+- **successes** - The number of successful checks in the form of an integer
+- **note** - Optional, a short note providing additional information about the test
+- **failures** - The number of failed checks in the form of an integer
+- **warnings** - The number of warning checks in the form of an integer
+
+Example contents of the test result output file (**HACBS_TEST_OUTPUT**) for a failed run:
+```
+{ 
+    "result": "FAILURE",
+    "namespace": "image_labels",
+    "timestamp": "1649148140",
+    "successes": 12, 
+    "note": "",
+    "failures": 2
+}
+```
+Example for a successful run:
+```
+{
+    "result": "SUCCESS",
+    "timestamp": "1649843611",
+    "namespace": "required_checks",
+    "successes": 16,
+    "note": "",
+    "failures": 0
+}
+```
+Example for a skipped run:
+```
+{
+    "result": "SKIPPED",
+    "note": "<skipped message>",
+    "timestamp": "1649842004"
+}
+```
+Example for a run with an error:
+```
+{
+    "result": "ERROR",
+    "timestamp": "1649842004"
+}
+```
+
+##### Proposed Adapted Tekton Result Format for Xunit Tests Results
+
+- Existing Mapped:
+
+  - **Namespace** == **name** attribute of all of the **TestSuite** elements 
```

Comment:
Keys are not capitalized above. Should we be consistent and not capitalize here?

Reply:
Updated. Thanks.